src/engine/source/api/include/api/test/sessionManager.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef _SESSIONS_MANAGER_HPP
+#define _SESSIONS_MANAGER_HPP
+
+#include "iSessionManager.hpp"
+
+#include <shared_mutex>
+#include <unordered_map>
+
+namespace api::sessionManager
+{
+
+class SessionManager : public ISessionManager
```

Comment:
improve-desing: the "final" operator can be placed to ensure that no other class will inherit from "SessionManager"